components/calendar/src/chinese.rs

```diff
@@ -448,6 +477,59 @@ impl Chinese {
             related_iso,
         }
     }
+
+    fn unpack_chinese_data(
+        year: i32,
+        packed_data: ChineseData,
+        next_year_data: ChineseData,
+    ) -> ChineseBasedCompiledData {
+        let new_year_offset = ((packed_data.0 & 0b11111000) >> 3) as u16;
+        let iso_year = 2023 + (year - 4660);
+        let new_year =
+            Iso::fixed_from_iso(Iso::iso_from_year_day(iso_year, 21 + new_year_offset).inner);
+
+        let next_new_year_offset = ((next_year_data.0 & 0b11111000) >> 3) as u16;
+        let next_iso_year = iso_year + 1;
+        let next_new_year = Iso::fixed_from_iso(
```

Comment:
A clever/faster way of doing this is instead `29 * 12 + (packed_data.0 & 0b111).count_ones() + packed_data.1.count_ones() + (packed_data.2 &0b11000000).count_ones()` and then add 29 if it's a leap month.